pypeit/deprecated/wvutils.py

```diff
@@ -86,3 +87,110 @@ def hist_wavedisp(waves, disps, dispbin=None, wavebin=None, scale=1.0, debug=Fal
+
+    return y1
+
+#This was JFHs keck_hires_dev development version to try to deal with the strongly saturated line regions
+def smooth_ceil_cont_hires_dev(inspec1, smooth, percent_ceil = None, use_raw_arc=False, sigdetect = 10.0, fwhm = 4.0,
+                     large_scale_fwhm_fact=20.0, large_scale_sigrej=20.0):
+
+    """  Utility routine to smooth and apply a ceiling to spectra
+
+    Args:
+        inspec1 (ndarray):
+          Input spectrum, shape = (nspec,)
+        smooth (int):
+          Number of pixels to smooth by
+        percent_ceil (float, optional):
+          Upper percentile threshold for thresholding positive and negative values
+        use_raw_arc (bool, optional):
+          If True, use the raw arc and do not continuum subtract. Default = False
+        sigdetect (float, optional):
+          Peak finding threshold which is used if continuum subtraction will occur (i.e. if use_raw_arc = False)
+        fwhm (float, optional):
+          Fwhm of arc lines for peak finding if continuum subtraction will occur (i.e. if use_raw_arc = False)
+        large_scale_fwhm_fact (float, optional):
+          Number of fwhms to use as the width of the running median to take out large scale features from saturated
+          lines.
+        large_scale_percentile (float, optional):
+          Percentile of large_scale_smoothed arc to set as the threshold above which the arc is masked to zero.
+
+    Returns:
+        y1_out (ndarray):
+          Spectrum with smoothing and ceiling applied, shape = (nspec,)
+
+    """
+
+    # ToDO can we improve the logic here. Technically if use_raw_arc = True and perecent_ceil=None
+    # we don't need to peak find or continuum subtract, but this makes the code pretty uggly.
+
+    # Run line detection to get the continuum subtracted arc
+    tampl1, tampl1_cont, tcent1, twid1, centerr1, w1, arc1, nsig1 = arc.detect_lines(inspec1, sigdetect=sigdetect, fwhm=fwhm)
+
+    if use_raw_arc:
+        ampl = tampl1
+        use_arc = inspec1
+    else:
+        ampl = tampl1_cont
+        use_arc = arc1
+
+    if percent_ceil is not None and (ampl.size > 0):
+        # If this is set, set a ceiling on the greater > 10sigma peaks
+        ceil_upper = np.percentile(ampl[ampl >= 0.0], percent_ceil) if np.any(ampl >= 0.0) else 0.0
+        # Set a lower ceiling on negative fluctuations
+        ceil_lower = np.percentile(ampl[ampl < 0.0], percent_ceil) if np.any(ampl < 0.0) else 0.0
+        spec1 = np.clip(use_arc, ceil_lower, ceil_upper)
+    else:
+        spec1 = np.copy(use_arc)
+
+    if smooth is not None:
+        y1 = scipy.ndimage.filters.gaussian_filter(spec1, smooth)
+    else:
+        y1 = np.copy(spec1)
+
+    # Mask out large scale features
+    y1_ls = utils.fast_running_median(y1, fwhm * large_scale_fwhm_fact)
+    mean, med, stddev = astropy.stats.sigma_clipped_stats(
+        y1_ls[y1_ls != 0.0], sigma_lower=3.0, sigma_upper=3.0, cenfunc='median', stdfunc=utils.nan_mad_std)
```

Comment:
calculate `y1_ls != 0.0` once instead of 3 times?

Reply:
It's deprecated.